src/main/java/com/github/manolo8/darkbot/extensions/plugins/PluginUpdater.java

```diff
@@ -0,0 +1,258 @@
+package com.github.manolo8.darkbot.extensions.plugins;
+
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.gui.plugins.PluginCard;
+import com.github.manolo8.darkbot.gui.plugins.PluginDisplay;
+import com.github.manolo8.darkbot.utils.FileUtils;
+import com.github.manolo8.darkbot.utils.I18n;
+import com.github.manolo8.darkbot.utils.Time;
+
+import javax.swing.*;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+//todoo i18n
+public class PluginUpdater {
+
+    public final List<PluginException> UPDATING_EXCEPTIONS = new ArrayList<>();
+    private long lastChecked = System.currentTimeMillis();
```

Comment:
```suggestion
    private volatile long lastChecked = System.currentTimeMillis();
```
This is used by different threads, make it volatile